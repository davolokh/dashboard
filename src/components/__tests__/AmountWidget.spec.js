import React from 'react';
import renderer from 'react-test-renderer';
import AmountWidget from '../AmountWidget';
import { knownAccounts } from '../../common/known_accounts';

describe('AmountWidget', () => {
  it(`renders component: loading`, () => {
    const component = renderer.create(<AmountWidget  />);
    expect(component.toJSON()).toMatchSnapshot();
    component.unmount();
  });

  it(`renders component: loaded, 0 amount`, () => {
    const component = renderer.create(<AmountWidget  />);
    const instance = component.getInstance();

    instance.setState({ loading: false, amount: 0 });
    component.update(<AmountWidget  />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.unmount();
  });

  it(`renders component: loaded, 100000 amount`, () => {
    const component = renderer.create(<AmountWidget  />);
    const instance = component.getInstance();

    instance.setState({ loading: false, amount: 100000 });
    component.update(<AmountWidget  />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.unmount();
  });

  it(`renders component: loaded, 1000000 amount`, () => {
    const component = renderer.create(<AmountWidget  />);
    const instance = component.getInstance();

    instance.setState({ loading: false, amount: 1000000 });
    component.update(<AmountWidget  />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.unmount();
  });

  it(`renders component: loaded, 1000000000 amount`, () => {
    const component = renderer.create(<AmountWidget  />);
    const instance = component.getInstance();

    instance.setState({ loading: false, amount: 1000000000 });
    component.update(<AmountWidget  />);
    
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    component.unmount();
  });
});