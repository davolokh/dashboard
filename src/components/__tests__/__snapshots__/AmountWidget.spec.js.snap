// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AmountWidget renders component: loaded, 0 amount 1`] = `
<div
  className="mui-panel "
>
  <div
    className="widget-name"
  />
  <div
    className="mui--text-display3 mui--text-center"
  >
    0
  </div>
  <div
    className="mui--text-caption mui--text-center"
  >
    <span>
      0.0000000
       
    </span>
  </div>
</div>
`;

exports[`AmountWidget renders component: loaded, 100000 amount 1`] = `
<div
  className="mui-panel "
>
  <div
    className="widget-name"
  />
  <div
    className="mui--text-display3 mui--text-center"
  >
    100k
  </div>
  <div
    className="mui--text-caption mui--text-center"
  >
    <span>
      100,000.0000000
       
    </span>
  </div>
</div>
`;

exports[`AmountWidget renders component: loaded, 1000000 amount 1`] = `
<div
  className="mui-panel "
>
  <div
    className="widget-name"
  />
  <div
    className="mui--text-display3 mui--text-center"
  >
    1M
  </div>
  <div
    className="mui--text-caption mui--text-center"
  >
    <span>
      1,000,000.0000000
       
    </span>
  </div>
</div>
`;

exports[`AmountWidget renders component: loaded, 1000000000 amount 1`] = `
<div
  className="mui-panel "
>
  <div
    className="widget-name"
  />
  <div
    className="mui--text-display3 mui--text-center"
  >
    1B
  </div>
  <div
    className="mui--text-caption mui--text-center"
  >
    <span>
      1,000,000,000.0000000
       
    </span>
  </div>
</div>
`;

exports[`AmountWidget renders component: loading 1`] = `
<div
  className="mui-panel "
>
  <div
    className="widget-name"
  />
  <div
    className="mui--text-display3 mui--text-center"
  >
    Loading...
  </div>
  <div
    className="mui--text-caption mui--text-center"
  >
    
  </div>
</div>
`;
